isisd/isis_nb_config.c

```diff
@@ -1521,7 +1521,9 @@ int isis_instance_segment_routing_msd_node_msd_modify(
 
 	area = nb_running_get_entry(args->dnode, NULL, true);
 	area->srdb.config.msd = yang_dnode_get_uint8(args->dnode, NULL);
-	isis_sr_cfg_msd_update(area);
+
+	/* Update and regenerate LSP */
+	lsp_regenerate_schedule(area, area->is_type, 0);
```

Comment:
Why remove `isis_sr_cfg_msd_update()`, which did exactly the same thing? In the future we might need to do other things when the MSD config changes, and I'd prefer to have that code in the _isis_sr.c_ file rather than in this file (which is already very long).

Reply:
Well, in this case, we have to also move call to `lsp_refresh_schedule()` from isis_nb_config.c for the same reason. Honestly, I don't see what we could do more with MSD. MSD is just a parameter that is advertise to indicate constraints on label stack for segment path. This is only of interest for PCE which will take this into account for path computation. If you setup locally a segment path, we need to take into that constraints. But, as this constraint is linked to hardware capacity, you will not change it when ISIS-SR is running.

An from a pure code readability, a function with only one instruction is trivial.

Reply:
Fair enough. Marking as resolved...